### DISCUSSION TITLE
Standardize Lua hooks

### DISCUSSION THREAD

**Discussion by scarf005** [cataclysmbn/Cataclysm-BN#3722] [General]
currently lua hooks re-use existing hooks like `examine_action` or `use_action`. however, it is not namespaced, thus will be hard to manage when more hooks are added. therefore, i suggest a new centralized namespace for hooks: `"hooks"`.

```json
{
  "type": "GENERIC",
  "hooks": {
    "on_use": "FOO",
    "on_examine": "BAR",
    "on_destruction": "BAZ",
    "on_drop": "QUX",
    "on_throw": "QUUX"
  }
}
```

**Comment by Vollch** [comment]
Dedicated namespace will be especially useful if we'll have custom hooks at some point. E.g. Sky Island could introduce "on_raid_started" hook, which can be used later on by other mods.

**Comment by Vollch** [comment]
Another things regarding hooks which is worth standardizing, it's where callbacks should be physically stored, and how they will be loaded in game. Once we'll have callbacks for mapgens, iexamine, mondeath, and all the things - how they going to be organized within data tree? Single .lua file might work for mods, but in base game it'll become too hard to navigate eventually. We're not dumping everything in single json after all.
